core/blockchain.go

```diff
@@ -223,6 +223,22 @@ func (bc *BlockChain) ValidateNewBlock(block *types.Block, address common.Addres
 	return nil
 }
 
+// IsEpochBlock returns whether this block is the first block of an epoch.
```

Comment:
I have a similar function  CheckEpochBlock in core/resharding.go. So we don't need to duplicate it

Reply:
I think the checking is better to be on blockchain level. So I switched the usages of yours to mine.